src/cplus_plugin/lib/constant_raster.py

```diff
@@ -0,0 +1,836 @@
+                else component.component_id
+            )
+            if metadata_id and component_name:
+                # Use descriptive filename with value and unit
+                output_filename = generate_constant_raster_filename(
+                    component_name, absolute_value, metadata_id
+                )
+            else:
+                # Fallback to old naming scheme
+                safe_name = component_name.replace(" ", "_").replace("/", "_")
+                output_filename = f"constant_raster_{safe_name}.tif"
+
+            output_path = os.path.join(session_dir, output_filename)
+
+            try:
+                # Create the constant raster only if skip_raster is False
+                if not collection.skip_raster:
+                    if feedback:
+                        component_name = (
+                            component.component.name
+                            if component.component
+                            and hasattr(component.component, "name")
+                            else component.component_id
+                        )
+                        feedback.pushInfo(f"Component: {component_name}")
+                        feedback.pushInfo(f"Absolute value: {absolute_value}")
+                        feedback.pushInfo(
+                            f"Normalization range: {collection.min_value} - {collection.max_value}"
+                        )
+                        feedback.pushInfo(f"Normalized value: {constant_value}")
+                        feedback.pushInfo(
+                            f"Creating raster with normalized value {constant_value} at {output_path}"
+                        )
+
+                    created_path = ConstantRasterProcessingUtils.create_constant_raster(
+                        value=constant_value,
+                        raster_context=context,
+                        output_path=output_path,
+                        processing_context=processing_context,
+                        feedback=feedback,
+                    )
+
+                    if feedback:
+                        feedback.pushInfo(f"Successfully created: {created_path}")
+                else:
+                    # Skip raster creation, just use the output path
+                    created_path = output_path
+                    log(
+                        f"Skipped raster creation for {output_filename} (skip_raster=True)",
+                        info=True,
+                    )
+
+                # Always save metadata file (even when skip_raster=True)
+                if metadata_id:
+                    try:
+                        component_name = (
+                            component.component.name
+                            if component.component
+                            and hasattr(component.component, "name")
+                            else component.component_id
+                        )
+                        meta_path = save_constant_raster_metadata(
```

Comment:
Too many args, can this info be encapsulated in a dataclass so that if there any info exclusions or inclusions then we just modify the dataclass and its use inside the function without also having to modify the params?